site/src/components/AppLink/AppLink.tsx

```diff
@@ -29,6 +30,11 @@ export const AppLink: FC<AppLinkProps> = ({
   workspace,
   agent,
 }) => {
+  const preferredProxy = usePreferredProxy()
+  const preferredPathBase = preferredProxy ? preferredProxy.path_app_url : ""
+  // Use the proxy host subdomain if it's configured.
+  appsHost = preferredProxy ? preferredProxy.wildcard_hostname : appsHost
```

Comment:
The appsHost prop should probably be removed, as well as the apps host endpoint.

Reply:
I'm not sure I follow

Reply:
See my comment about what the context should do below. Instead of this component calling the `/api/v2/applications/host` route to get the primary wildcard host it should only use the `/api/v2/regions` endpoint, and the old legacy route should be removed

Reply:
Will simplify this later.